### PR TITLE
Enable template arg

### PR DESCRIPTION
**Pull request by vividviolet**
Fixes https://github.com/Shopify/argo-admin-private/issues/1369

### CODE REVIEW

**Review by henrytao-me — APPROVED** [review]
🎩-ed. LGTM

**Review by henrytao-me — COMMENTED** [review]
Checkout asking for these options 

```
Henrys-MacBook-Pro:argo-checkout-template henrytao$ yarn generate --type=CHECKOUT_POST_PURCHASE
yarn run v1.22.4
$ ts-node ./scripts/generate --type=CHECKOUT_POST_PURCHASE
Create  CHECKOUT_POST_PURCHASE  extension project
? Select template: 
  vanilla 
  react 
  vanilla-typescript 
❯ react-typescript 
```

while Admin asking for these

```
Henrys-MacBook-Pro:argo-admin-template henrytao$ yarn generate --type=PRODUCT_SUBSCRIPTION --template=
yarn run v1.22.4
$ ts-node -O "{\"esModuleInterop\": true}" ./scripts/generate --type=PRODUCT_SUBSCRIPTION --template=
Create PRODUCT_SUBSCRIPTION extension project
? Select template: (Use arrow keys)
❯ Vanilla JS 
  React 
  Vanilla JS with Typescript 
  React with Typescript 
```

I think Checkout version is better and we should use it these values for the argument `'vanilla', 'react', 'vanilla-typescript', 'react-typescript'`

cc @vividviolet

**Comment by vividviolet** [comment]
@henrytao-me I think the options presented in the prompt should be a human readable string which can be more descriptive. Is there a reason why you prefer the option be shorter ones like check out has it?

For the template argument, it could be shorter to make it easier to type.  @t6d and I were thinking that the new format is more consistent and makes it more extensible in the future. For example, we could have something like `js+react+graphql` as a template. We don't have to use the `+` but I think we should be consistent in the order, language first, then libraries/ frameworks after.

Thoughts @iainmcampbell @Fionoble?

**Comment by henrytao-me** [comment]
> Is there a reason why you prefer the option be shorter ones like check out has it?

I like what you see is what you get style. They see the options from inquirer, then they can use it for argument.

File: scripts/generate/generate-src.ts [hunk 1]
```diff
@@ -4,10 +4,10 @@ import path from 'path';
...
 
 export enum Template {
-  Vanilla = 'vanilla',
-  React = 'react',
-  VanillaTypescript = 'vanilla-typescript',
-  ReactTypescript = 'react-typescript',
+  Vanilla = 'js',
+  React = 'js+react',
+  VanillaTypescript = 'ts',
+  ReactTypescript = 'ts+react',
```
Comment (Fionoble) [inline scripts/generate/generate-src.ts:10]: I'm not super sold on the `+` separator but I have absolutely no valid reasoning for it. However I do like adding the more structured approach with language then framework 👍

**Review by Fionoble — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -19,7 +19,7 @@ npm run generate -- --type=PRODUCT_SUBSCRIPTION
 ## With yarn
 yarn
 
-yarn generate --type=PRODUCT_SUBSCRIPTION
+yarn generate --type=PRODUCT_SUBSCRIPTION --template=js+react
```
Comment (vividviolet) [inline README.md:22]: Can you add a section that lists out the available templates similar to the `Available extensions` section?

**Review by t6d — APPROVED** [review]

**Comment by henrytao-me** [comment]
Hmmmmmm. Can we think about the more descriptive vs shorthand again?

**Comment by henrytao-me** [comment]
Discussed offline, we will go with these options for both argument and picker

```
javascript
typescript
javascript-react
typescript-react
```

**Pull request merged**